### PR TITLE
Stop using :not() pseudo class for mx_GenericEventListSummary

### PR DESCRIPTION
**Pull request by luixxiul**
This PR removes `:not()` pseudo class from `mx_GenericEventListSummary`, removing the possibility of unexpected regressions and fixing the inline start (left) padding on IRC layout at the same time.

Though there are other issues to be fixed there (such as block padding around info lines), fixing them is a big task and should be deferred to another PR.

Closes https://github.com/vector-im/element-web/issues/22602

|Before|After|
|---------|------|
|before1|after1|

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

type: defect

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Stop using :not() pseudo class for mx_GenericEventListSummary ([\#8944](https://github.com/matrix-org/matrix-react-sdk/pull/8944)). Fixes vector-im/element-web#22602. Contributed by @luixxiul.

### CODE REVIEW

File: res/css/views/rooms/_IRCLayout.scss [hunk 1]
```diff
@@ -152,26 +152,23 @@ $irc-line-height: $font-18px;
...
-    .mx_GenericEventListSummary > .mx_EventTile_line {
-        padding-left: calc(var(--name-width) + $icon-width + $MessageTimestamp_width + 3 * $right-padding); // 15 px of padding
-    }
-
     .mx_EventTile.mx_EventTile_info {
         .mx_EventTile_avatar {
-            left: calc(var(--name-width) + 10px + $icon-width);
+            left: calc(var(--name-width) + 10px + var(--icon-width));
             top: 0;
+            margin-right: var(--right-padding);
```
Comment (luixxiul) [inline res/css/views/rooms/_IRCLayout.scss:167]: This creates gap between the avatar and the info line.

**Comment by dbkr** [comment]
If this is fixing a bug, it should be defect rather than task

**Comment by dbkr** [comment]
although I'm not sure I have enough context to give a sensible review on this I'm afraid

**Review by dbkr — APPROVED** [review]

**Comment by luixxiul** [comment]
@dbkr Thanks for reviewing.

**Pull request merged**